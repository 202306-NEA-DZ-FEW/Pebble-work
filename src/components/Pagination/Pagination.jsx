import { useState } from "react";

export const usePagination = (initialPage, initialItemsPerPage, items) => {
    const [currentPage, setCurrentPage] = useState(initialPage);
    const [itemsPerPage, setItemsPerPage] = useState(initialItemsPerPage);

    const totalPages = Math.ceil(items.length / itemsPerPage);
    const indexOfLastItem = currentPage * itemsPerPage;
    const indexOfFirstItem = indexOfLastItem - itemsPerPage;
    const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);

    return {
        currentPage,
        itemsPerPage,
        totalPages,
        currentItems,
        setCurrentPage,
        setItemsPerPage,
    };
};
