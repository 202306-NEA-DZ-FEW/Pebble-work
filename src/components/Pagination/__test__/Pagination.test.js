import React from "react";
import { render, screen } from "@testing-library/react";
import { usePagination } from "../Pagination";

// A simple component that uses the hook
function TestComponent() {
    const items = ["Item 1", "Item 2", "Item 3", "Item 4", "Item 5"];
    const {
        currentPage,
        itemsPerPage,
        totalPages,
        currentItems,
        setCurrentPage,
        setItemsPerPage,
    } = usePagination(1, 2, items);

    return (
        <div>
            {currentItems.map((item, index) => (
                <div key={index}>{item}</div>
            ))}
        </div>
    );
}

describe("usePagination", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<TestComponent />);
        expect(asFragment()).toMatchSnapshot();
    });
});
