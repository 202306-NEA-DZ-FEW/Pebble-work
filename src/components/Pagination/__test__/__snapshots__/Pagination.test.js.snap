// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePagination renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
  </div>
</DocumentFragment>
`;
